src/main/drivers/accgyro/accgyro_spi_icm426xx.c

```diff
@@ -186,65 +206,35 @@ void icm426xxGyroInit(gyroDev_t *gyro)
     spiWriteReg(dev, ICM426XX_RA_PWR_MGMT0, ICM426XX_PWR_MGMT0_TEMP_DISABLE_OFF | ICM426XX_PWR_MGMT0_ACCEL_MODE_LN | ICM426XX_PWR_MGMT0_GYRO_MODE_LN);
     delay(15);
 
-    uint8_t odrIdx = 0;
-    bool supportedODRFound = false;
-
-    if (gyro->gyroRateKHz) {
-        const uint8_t gyroSyncDenominator = gyro->mpuDividerDrops + 1; // rebuild it in here, see gyro_sync.c
-        const uint8_t desiredODR = 8 / gyroSyncDenominator;
-        for (unsigned i = 0; i < ARRAYLEN(odrConfigLUT); i++) {
-            if (odrConfigLUT[i].odr == desiredODR) {
-                odrIdx = odrConfigLUT[i].odr_idx;
-                supportedODRFound = true;
-                break;
-            }
-        }
-    }
-
-    if (!supportedODRFound) {
-        odrIdx = 6;
+    // Get desired output data rate
+    uint8_t odrConfig;
+    const uint8_t decim = LOG2_8BIT(gyro->mpuDividerDrops + 1);
```

Comment:
LOG2_8BIT is quite bad idea - it is intended for compile-time constants (generated code will be quite large). If you really want to to compute it, use `__builtin_clz` (possibly implementing `llog2` in utils.h).
Or 
```
unsigned decim = 0;
for(unsigned d = gyro->mpuDividerDrops; d; d >>= 1) {
  decim++;
}
```
But ODR register isn't actually log divider on this gyro - 500Hz is missing and next is 200Hz, so this code will work only for >= 1kHz

Personally, I'd prefer 8bytes long table in this case. Aut this code is unused anyway ..

Reply:
I would have guessed that the compiler already would have broken this down into a compile time constant because I assumed that it detected `gyro->mpuDividerDrops` being referentially transparent (always 0). But maybe not, because your suggested implementation using `__builtin_clz` brought the flash size down a bit (-16 bytes).

Do/did we even care about gyro rates below 1000Hz? Since we are not going below 500Hz we can utilize the "power of 2 pattern", I reckon. We could include 500Hz as the next log divider but even slower rates don't make any sense and would be overengineering at this point.

Reply:
Predicting variable value in C isn't that simple. Roughly speaking, aby globally reachable variable can be changed inside function.

The is to 500Hz for this gyro - next step is 200Hz. 

Just get finished with it, maybe noting 1k limit in comments